pkg/frontend/querymiddleware/roundtrip_test.go

```diff
@@ -255,12 +255,16 @@ func TestTripperware_InstantQuery(t *testing.T) {
 
 func TestTripperware_Metrics(t *testing.T) {
 	tests := map[string]struct {
-		path             string
+		request          *http.Request
 		stepAlignEnabled bool
 		expectedMetrics  string
 	}{
 		"range query, start/end is aligned to step": {
-			path: "/api/v1/query_range?query=up&start=1536673680&end=1536716880&step=120",
+			request: func() *http.Request {
+				req, err := http.NewRequest("GET", "/api/v1/query_range?query=up&start=1536673680&end=1536716880&step=120", http.NoBody)
+				require.NoError(t, err)
+				return req
+			}(),
```

Comment:
Nit, httptest is already imported, you can just do 

```suggestion
			request: httptest.NewRequest("GET", "/api/v1/query_range?query=up&start=1536673680&end=1536716880&step=120", nil),
```

and in other places bellow.

Reply:
TIL, done in https://github.com/grafana/mimir/pull/8413/commits/fd91434bf8f47d8c8a601f5521ed05624ee32278